nautobot_data_validation_engine/jobs.py

```diff
@@ -0,0 +1,85 @@
+"""Jobs for nautobot_data_validation_engine."""
+
+from django.apps import apps as global_apps
+
+from nautobot.extras.models import GitRepository
+from nautobot.extras.jobs import Job, MultiChoiceVar
+
+from nautobot_data_validation_engine.custom_validators import get_data_compliance_rules_map, get_classes_from_git_repo
+from nautobot_data_validation_engine.models import DataCompliance
+
+
+def get_data_compliance_rules():
+    """Generate a list of Audit Ruleset classes that exist from the registry."""
+    validators = []
+    for rule_sets in get_data_compliance_rules_map().values():
+        validators.extend(rule_sets)
+    return validators
+
+
+def get_choices():
+    """Get choices from registry."""
+    choices = []
+    for ruleset_class in get_data_compliance_rules():
+        choices.append((ruleset_class.__name__, ruleset_class.__name__))
+    for repo in GitRepository.objects.all():
+        if "nautobot_data_validation_engine.data_compliance_rules" in repo.provided_contents:
+            for compliance_class in get_classes_from_git_repo(repo):
+                choices.append((compliance_class.__name__, compliance_class.__name__))
+
+    choices.sort()
+    return choices
+
+
+class RunRegisteredDataComplianceRules(Job):
+    """Run the validate function on all registered DataComplianceRule classes."""
+
+    name = "Run Registered Data Compliance Rules"
+
+    selected_data_compliance_rules = MultiChoiceVar(
+        choices=get_choices,
+        label="Select Data Compliance Rules",
+        required=False,
+        description="Not selecting any rules will run all rules listed.",
+    )
+
+    def run(self, data, commit):
+        """Run the validate function on all given DataComplianceRule classes."""
+        selected_data_compliance_rules = data.get("selected_data_compliance_rules")
+
+        compliance_classes = []
+        compliance_classes.extend(get_data_compliance_rules())
+
+        for repo in GitRepository.objects.all():
+            if "nautobot_data_validation_engine.data_compliance_rules" in repo.provided_contents:
+                compliance_classes.extend(get_classes_from_git_repo(repo))
+
+        for compliance_class in compliance_classes:
+            if selected_data_compliance_rules and compliance_class.__name__ not in selected_data_compliance_rules:
+                continue
+            self.log_info(f"Running {compliance_class.__name__}")
+            app_label, model = compliance_class.model.split(".")
+            for obj in global_apps.get_model(app_label, model).objects.all():
+                ins = compliance_class(obj)
+                ins.enforce = False
+                ins.clean()
+
+
+class DeleteOrphanedDataComplianceData(Job):
```

Comment:
i think orphaned or dependent is the right relationship forming word there… `DataCompliance` objects are only created because of other objects so i think it’s important to show that relation when we’re trying to delete these objects.  i feel that orphaned shows the stronger relation of the two but will admit that it’s not a direct parent-child relation

Reply:
quick google search seems to show that orphaned is much more associated with django than dangling.